src/server/scripts/ExilesReach/zone_exiles_reach.cpp

```diff
@@ -3805,6 +3805,290 @@ class spell_scene_linger_northbound: public SpellScript
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                {
+                    me->SetFacingToObject(player);
+                    me->SetEmoteState(EMOTE_STATE_TALK);
+                    Talk(SAY_PET_TRAINING_ALLIANCE);
+                    player->CastSpell(player, SPELL_LEARNING_TAME_BEAST);
+                    _events.ScheduleEvent(EVENT_ME_END_OF_CAST, 8s);
+                }
+                break;
+            case EVENT_ME_END_OF_CAST:
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                {
+                    if (player->GetQuestStatus(QUEST_TRACKER_TAMING_THE_WILDS_COMPLETE) != QUEST_STATUS_REWARDED)
+                    {
+                        player->CastSpell(player, SPELL_TUTORIAL_HEALTH_DNT);
+                    }
+
+                    if (!player->HasSpell(SPELL_TAME_BEAST))
+                    {
+                        player->UpdateQuestObjectiveProgress(QUEST_OBJECTIVE_CRITERIA_TREE, QUEST_OBJECTIVE_TRAINED_ALLIANCE, 1);
+                        player->CastSpell(player, SPELL_LEARN_TAME_BEAST);
+                        player->CastSpell(player, SPELL_LEARN_CALL_PET);
+                    }
+
+                    me->SetEmoteState(EMOTE_STATE_NONE);
+                    Talk(SAY_FIND_A_BEAST_ALLIANCE);
+                }
+                me->DespawnOrUnsummon(4s);
+                break;
+            default:
+                break;
+            }
+        }
+    }
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+};
+
+// This script is used by Austin Huxworth for Taming The Wild quest
+struct npc_dawntracker_hunter_quest_private : public ScriptedAI
+{
+    npc_dawntracker_hunter_quest_private(Creature* creature) : ScriptedAI(creature) { }
+
+    void InitializeAI() override
+    {
+        me->RemoveNpcFlag(UNIT_NPC_FLAG_QUESTGIVER | UNIT_NPC_FLAG_GOSSIP);
+    }
+
+    void JustAppeared() override
+    {
+        me->SetStandState(UNIT_STAND_STATE_STAND);
+        _events.ScheduleEvent(EVENT_ME_TURN_TO_PLAYER, 1s);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        Player* player = summoner->ToPlayer();
+        if (!player)
+            return;
+
+        _playerGUID = player->GetGUID();
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+            case EVENT_ME_TURN_TO_PLAYER:
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                {
+                    me->SetFacingToObject(player);
+                    me->SetEmoteState(EMOTE_STATE_TALK);
+                    Conversation* conversation = Conversation::CreateConversation(CONVERSATION_HORDE_TRAINER, player, *player, player->GetGUID(), nullptr, false);
+                    conversation->AddActor(CONVERSATION_HORDE_ACTOR, 0, me->GetGUID());
```

Comment:
missing nullcheck for conversation (possible crash)

Reply:
Done